### ISSUE TITLE
RE: Alpha Variants for Black and White 

### ISSUE THREAD

**Issue by lafllamme** [endigma/unocss-preset-radix#26]
Hi,  
First of all, I want to apologize for reopening this issue. Referring to the previous issue: https://github.com/endigma/unocss-preset-radix/issues/25, I initially thought the problem was resolved. However, I’ve realized the issue is not only unresolved, but it seems to have worsened.

Currently, I’m encountering the following problems:  

1. **No Black or White Colors Rendered Properly**:  
   - In my IDE, I normally get a color preview for all colors when I use the syntax like `bg-color-scale` or `bg-color-scaleA` (for alpha).  
   - When I use the same syntax for `bg-black` or `bg-white`, I don’t get a preview. Additionally, the colors don’t render correctly in the browser.  
   - First I though it's dependent to the Plugin within my IDE and tried both variants (regular and the color-blackA-12)
   
   
![Regular](https://github.com/user-attachments/assets/b356865c-f92a-4ffa-a2ed-73cfe3383369)

2. **Alternative Syntax Behavior**:  
   - If I use `bg-blackA-scale` (e.g., `bg-blackA-11`), the IDE gives me a preview, but the scales don’t seem correct.  
   - Even with `bg-blackA-scale`, the color variable is applied in the browser but doesn’t actually set the color.
   
   ![Other writewise](https://github.com/user-attachments/assets/1e412592-889e-4ae3-bca3-b9ff472f725f)  

3. **Mapping and Syntax Issues**:  
   - Looking at my UnoCSS setup, I only see `blackA` but not all the scales.  
   - It seems there are no proper mappings for black and white colors with alpha scales, or they aren’t being set correctly.  
   

---

### Screenshots:  
I’ve attached a screenshot showing how the variables are applied in the browser with regular write wise (e.g. bg-black-6A), suggested/rendered one (e.g. bg-blackA-6) but fail to set the actual color-variable.

![Regular](https://github.com/user-attachments/assets/78ec3bcf-dd0a-4723-8943-5aa3ae8706e7)
![Other writewise](https://github.com/user-attachments/assets/f0f59c84-49ae-4824-af0a-0b633a68dbd2)
![Rendred File](https://github.com/user-attachments/assets/3b594089-e351-468c-a1d6-dc2595cf7ea4)

---

### Steps to Reproduce:  
1. Use the syntax `bg-black-scale` or `bg-black-scaleA` in your project.  
2. Observe that no preview is shown in the IDE.  
3. Check the browser and see that the color variable is applied but not rendered.  
4. Test alternative syntax like `bg-blackA-scale`, which provides partial previews in the IDE but fails to render properly in the browser.  

---

### Expected Behavior:  
- `bg-black` and `bg-white` (with or without alpha scales) should preview correctly in the IDE and render properly in the browser.  
- All alpha scales for black and white should be available and functional.  

---

### Additional Notes:  
This time, I’ll make sure to test thoroughly before assuming the issue is resolved. I didn’t expect the previous fix to cause further issues, and I appreciate your patience while I work through this.  

Thanks for your support and Happy New Year🎉!

**Comment by endigma** [comment]
Uh, yikes. I'll look into this tomorrow, it seemed to work for me in the docs project.

**Comment by endigma** [comment]
I can't reproduce this. `bg-black(\d{1,2})A` all work correctly in the docs. I don't know why the IDE integration isn't picking this up properly, but that's the correct syntax as it is for all the other colors. You may need to add `"black"` to your palette now that it's a proper scale.

**Comment by endigma** [comment]
2.5.7 reintroduces the basic `black` and `white` colors which can be used with opacity the standard way.

**Comment by lafllamme** [comment]
@endigma You're absolutely right—thank you for pointing that out! I didn’t initially think of it as a color itself, so I forgot to include it in my UnoCSS configuration.

Now, I’m getting the black-scales and white-scales as expected, and I can verify the changes.

Can i tip you a coffee somewhere? Let me know

Thanks so much for your effort, and Happy New Year!

**Comment by lafllamme** [comment]
I noticed something interesting: with version `2.5.6`, I can see the palettes without any issues, but with `2.5.7`, they no longer appear. If you'd like, you can check out my project on GitHub to test and reproduce the issue: [[Nuxt-Design-System](https://github.com/lafllamme/Nuxt-Design-System)](https://github.com/lafllamme/Nuxt-Design-System).

Getting it up and running is super easy—just clone the repo, and use `make upp`. The Docker container will handle everything for you. Once it’s ready, head over to `localhost:6006`. There's a `ThemeToggle` component under Storybook that showcases all the colors, so it should be straightforward to test.

Thanks again for your help, and feel free to reach out if you have any questions. Happy New Year!

**Comment by endigma** [comment]
Can you clarify which tokens are or are not working? I have my own test scenarios for this plugin I've been updating as I go.

**Comment by endigma** [comment]
Hey @lafllamme did you resolve this? I'm not sure what you're saying doesn't work, and out of principle I generally don't download and run code I haven't read on my computer. Screenshots, a codepen, stackblitz, etc are all good if you're still experiencing the issue.

**Comment by lafllamme** [comment]
Hey @endigma,

Apologies for the delay! I’ve been busy assembling and upgrading parts for my new PC. The setup is complete now, and I’ve got a powerful new workstation!

Regarding your question, I’ve prepared a small example for the color palette on StackBlitz:

[Stackblitz Demo](https://stackblitz.com/edit/nuxt-starter-zvjgwb89?file=app.vue)

Feel free to modify and adjust it as needed. I’m not sure if I’ve missed anything, but this setup is essentially a 1:1 replica of what I have in my Storybook. The issue I mentioned earlier is also happening here.

The only special thing I’m doing is dynamically generating the classes, but that shouldn't affect anything since I’m using a safelist. I’ve added console logs throughout the code so you can take a closer look. The safelisting itself seems fine—it works with the regular colors, so I don’t see why it wouldn’t work for black/white.

One strange behavior I noticed is this:

With version `2.7.0` of the package, I get the regular black/white colors but no alpha scales. The same issue occurs with version `2.5.7.`

![Version_2 5 6](https://github.com/user-attachments/assets/9e615830-f224-404c-99f3-d426231cd485)
![Version_2 6 0](https://github.com/user-attachments/assets/47c6a7bb-f739-46aa-9f5f-278aca1a9d0f)


I added two `package.json` files within the demo:
one working (with `package.json.working` for scales, version `2.5.6`) and one regular (just `package.json`—you can rename or delete it). 

However, with version `2.5.6`, I have the black and white scales available, but no black or white as a color itself.

This issue is actually preventing me from progressing with my design system.

By the way, I’ve tipped you a big coffee as a small thank you for your hard work, especially for your quick responses. I really appreciate it!

Thanks for taking a look, and feel free to reach out if you need any further details!

**Comment by endigma** [comment]
Ok I messed up the scales with the explicit black/white values. I'm not sure there's a way around this unfortunately, as typescript doesn't allow something to have multiple values for one key, which is what would be required to have both bg-black and bg-black-XA work. I'll revert the update.

**Comment by endigma** [comment]
I've released 2.7.0 which removes the simple `black` and `white`. These names cannot conflict, so I'm leaving it up to the user to merge in their own theme if these are required, you can do something like "purewhite" or "pureblack" in the project config to add them back with whatever names you find acceptable.

**Comment by endigma** [comment]
That should work unless there's some gremlin in config merging I'm unaware of. I've extended this preset with brand colors before in personal projects. 

I'm not going to introduce more scale names on top of the official Radix colors in this preset, I recommend extending with whatever name you're happy with here. There _are_ pure black and white in the Radix scales as `12A`.
